libraries/ts-command-line/src/parameters/CommandLineDefinition.ts

```diff
@@ -62,6 +62,11 @@ export interface IBaseCommandLineDefinition {
    *   ordinary String Parameter:  Any value is accepted, including an empty string.
    */
   environmentVariable?: string;
+
+  /**
+   * Custom tab completions for the parameter values
+   */
+  completions?: () => Promise<string[]>;
 }
```

Comment:
Adding `completions` to `IBaseCommandLineDefinition` makes this option available to some parameter types that don't really make sense:
- `CommandLineFlagParameter`
- `CommandLineIntegerParameter`
Should we be more restrictive?

Reply:
Good point. Fixed.